### ISSUE TITLE
Nil pointer in Turtle::Writer#preprocess_statement

### ISSUE THREAD

**Issue by jcoyne** [ruby-rdf/rdf-turtle#11]
```
g = RDF::Graph.new
g.insert([RDF::URI.new('http://example.com/x'), RDF::DC.title, nil])
g.dump(:ttl)

NoMethodError: undefined method `literal?' for nil:NilClass
	from /Users/justin/.rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rdf-turtle-1.1.4/lib/rdf/turtle/writer.rb:378:in `preprocess_statement'
	from /Users/justin/.rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rdf-turtle-1.1.4/lib/rdf/turtle/writer.rb:359:in `block in preprocess'
	from /Users/justin/.rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rdf-1.1.4.2/lib/rdf/repository.rb:341:in `call'
	from /Users/justin/.rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rdf-1.1.4.2/lib/rdf/repository.rb:341:in `block (4 levels) in query_pattern'
	from /Users/justin/.rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rdf-1.1.4.2/lib/rdf/repository.rb:339:in `each'
	from /Users/justin/.rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rdf-1.1.4.2/lib/rdf/repository.rb:339:in `block (3 levels) in query_pattern'
	from /Users/justin/.rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rdf-1.1.4.2/lib/rdf/repository.rb:336:in `each'
	from /Users/justin/.rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rdf-1.1.4.2/lib/rdf/repository.rb:336:in `block (2 levels) in query_pattern'
	from /Users/justin/.rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rdf-1.1.4.2/lib/rdf/repository.rb:333:in `each'
	from /Users/justin/.rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rdf-1.1.4.2/lib/rdf/repository.rb:333:in `block in query_pattern'
	from /Users/justin/.rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rdf-1.1.4.2/lib/rdf/repository.rb:330:in `each'
	from /Users/justin/.rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rdf-1.1.4.2/lib/rdf/repository.rb:330:in `query_pattern'
	from /Users/justin/.rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rdf-1.1.4.2/lib/rdf/mixin/queryable.rb:83:in `query'
	from /Users/justin/.rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rdf-1.1.4.2/lib/rdf/model/graph.rb:248:in `each'
	from /Users/justin/.rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rdf-turtle-1.1.4/lib/rdf/turtle/writer.rb:359:in `preprocess'
	from /Users/justin/.rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rdf-turtle-1.1.4/lib/rdf/turtle/writer.rb:161:in `write_epilogue'
... 5 levels...
	from /Users/justin/.rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rdf-1.1.4.2/lib/rdf/writer.rb:150:in `buffer'
	from /Users/justin/.rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rdf-1.1.4.2/lib/rdf/writer.rb:130:in `dump'
	from /Users/justin/.rbenv/versions/2.1.1/lib/ruby/gems/2.1.0/gems/rdf-1.1.4.2/lib/rdf/mixin/enumerable.rb:678:in `dump'
	from (irb):5
```

**Comment by gkellogg** [comment]
Fixed in ruby-rdf/rdf/issues#226; incomplete statements always raise an error now, not just when validating.

**Comment by jcoyne** [comment]
Cool!
